assets/js/admin/admin.js

```diff
@@ -16,6 +21,73 @@ jQuery( document ).ready( function( $ ) {
 		}
 	} );
 
+	/**
+	 * Populate and enable the select element for days of the month.
+	 *
+	 * @param {integer} year
+	 * @param {integer} month
+	 * @param {jQuery} $day
+	 */
+	function loadDays( year, month, $day ) {
+		const dayCount = daysInMonth( year, month );
+		const dayVal = $day.val();
+		$day.children( 'option' ).remove();
+		const option = document.createElement( 'option' );
+		option.setAttribute( 'value', '' );
+		option.innerText = __( 'None', 'learning-commons-framework' );
+		$day.append( option );
+		for ( let i = 1; i < dayCount + 1; i++ ) {
+			const option = document.createElement( 'option' );
+			const val = 9 > i ? `0${i}` : i;
+			option.setAttribute( 'value', val );
+			option.innerText = i;
+			$day.append( option );
+		}
+		if ( dayVal < dayCount ) {
+			$day.val( dayVal );
+			$day.parents( '.cmb-row' ).removeClass( 'form-invalid' );
+			$day.siblings( '.error' ).remove();
+		} else {
+			$day.val( '' );
+			$day.parents( '.cmb-row' ).addClass( 'form-invalid' );
+			const errorText = __( 'The previously selected publication day is not valid in combination with the new year and month.', 'learning-commons-framework' );
+			const error = $( `<p class="error">${errorText}</p>` );
+			$day.siblings( '.cmb2-metabox-description' ).after( error );
+			speak( errorText );
+		}
+	}
+
+	$year.change( ( e ) => {
+		let yearVal = $( e.target ).val();
+		if ( ! yearVal ) {
+			yearVal = new Date().getFullYear();
+		}
+		let monthVal = $month.val();
+		if ( ! monthVal ) {
+			monthVal = new Date().getMonth();
+
+		}
+		loadDays( yearVal, monthVal, $day );
+	} );
+
+	$month.change( ( e ) => {
```

Comment:
If you change the month value to "None", the day should also change to "None". Otherwise it is possible to submit a resource with a year and day, but no month.

Reply:
Fixed in bacbbdf.